__tests__/distributors/base.test.ts

```diff
@@ -0,0 +1,180 @@
+import * as tc from '@actions/tool-cache';
+import * as core from '@actions/core';
+
+import path from 'path';
+import * as semver from 'semver';
+
+import { JavaBase } from '../../src/distributors/base-installer';
+import {
+  JavaDownloadRelease,
+  JavaInstallerOptions,
+  JavaInstallerResults
+} from '../../src/distributors/base-models';
+
+class EmptyJavaBase extends JavaBase {
+  constructor(installerOptions: JavaInstallerOptions) {
+    super('Empty', installerOptions);
+  }
+
+  protected downloadTool(javaRelease: JavaDownloadRelease): Promise<JavaInstallerResults> {
+    throw new Error('Method not implemented.');
+  }
+
+  protected findPackageForDownload(range: semver.Range): Promise<JavaDownloadRelease> {
+    throw new Error('Method not implemented.');
+  }
+}
+
+describe('Test findInToolcache', () => {
+  const actualJavaVersion = '11.1.10';
+  const javaPath = path.join('Java_Empty_jdk', actualJavaVersion, 'x86');
+
+  let mockJavaBase: EmptyJavaBase;
+  let tcFind: jest.SpyInstance;
+
+  beforeEach(() => {
+    tcFind = jest.spyOn(tc, 'find');
+    tcFind.mockImplementation((toolname: string, javaVersion: string, architecture: string) => {
+      const semverVersion = new semver.Range(javaVersion);
+
+      if (path.basename(javaPath) !== architecture || !javaPath.includes(toolname)) {
+        return '';
+      }
+
+      return semver.satisfies(actualJavaVersion, semverVersion) ? javaPath : '';
+    });
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '11', arch: 'x86', packageType: 'jdk' },
+      { javaVersion: actualJavaVersion, javaPath }
+    ],
+    [
+      { version: '11.1', arch: 'x86', packageType: 'jdk' },
+      { javaVersion: actualJavaVersion, javaPath }
+    ],
+    [{ version: '11', arch: 'x86', packageType: 'jre' }, null],
+    [{ version: '8', arch: 'x86', packageType: 'jdk' }, null],
+    [
+      { version: '11.1.10', arch: 'x86', packageType: 'jdk' },
+      { javaVersion: actualJavaVersion, javaPath }
+    ],
```

Comment:
nitpick: sorting, put all non-null cases at top.

Reply:
Done. Thank you.